modin/pandas/base.py

```diff
@@ -925,19 +925,69 @@ def at_time(self, time, asof=False, axis=None):  # noqa: PR01, RT01, D200
         """
         Select values at particular time of day (e.g., 9:30AM).
         """
-        axis = self._get_axis_number(axis)
-        idx = self.index if axis == 0 else self.columns
-        indexer = pandas.Series(index=idx).at_time(time, asof=asof).index
-        return self.loc[indexer] if axis == 0 else self.loc[:, indexer]
+        # TODO: before merging with master, add a base query compiler at_time that
+        # does the old implementation of indexing with a pandas series.
+        if asof:
+            # pandas raises NotImplementedError for asof=True, so we do, too.
+            raise NotImplementedError("'asof' argument is not supported")
+        return self.between_time(
+            start_time=time, end_time=time, inclusive="both", axis=axis
+        )
 
     @_inherit_docstrings(
         pandas.DataFrame.between_time, apilink="pandas.DataFrame.between_time"
     )
-    def _between_time(self, **kwargs):
-        axis = self._get_axis_number(kwargs.pop("axis", None))
-        idx = self.index if axis == 0 else self.columns
-        indexer = pandas.Series(index=idx).between_time(**kwargs).index
-        return self.loc[indexer] if axis == 0 else self.loc[:, indexer]
+    def _between_time(
+        self: "BasePandasDataset",
+        start_time,
+        end_time,
+        include_start: "bool_t | NoDefault",
+        include_end: "bool_t | NoDefault",
+        inclusive: "str | None",
+        axis,
+    ):
+        # TODO: before merging with master, add a base query compiler between_time that
+        # does the old implementation of indexing with a pandas series.
+
+        old_include_arg_used = (include_start is not no_default) or (
+            include_end is not no_default
+        )
+
+        if old_include_arg_used and inclusive is not None:
+            raise ValueError(
+                "Deprecated arguments `include_start` and `include_end` "
+                "cannot be passed if `inclusive` has been given."
+            )
```

Comment:
are we matching pandas behavior here? I'm assuming so, but just confirming

Reply:
Yep, all this error checking is copied from pandas.